### ISSUE TITLE
Project Submission

### ISSUE THREAD

**Issue by CodeMaxx** [CodeMaxx/my-other-computer-is-your-computer#1]
# Deadline is 4th night
## Preprocessing
- [x] Exploratory data analysis - Check the frequency of different kinds of data
- [x] Dump the features of every file you process so you don't have to compute them again and again -- This way we can also use all 20000 samples for training rather than 3500
- [x] Keep about 30% files as final untouched testcases

## Models to Try
- [x] SVM
- [x] adaboost/xgboost
- [x] Logistic
- [x] Neural Network with less features
- [ ] Semi-supervised learning with less data
- [x] kNeighbours
- [x] Random forest
- [ ] Try Deep learning on the feature dumps directly??
- [x] Hyperparameter tuning - SVM
- [x] Hyperparameter tuning - xgboost
- [x] Hyperparameter tuning - logistic
- [x] Hyperparameter tuning - neural
- [ ] Hyperparameter tuning - semi super
- [x] Hyperparameter tuning - k neighbour
- [x] Hyperparameter tuning - random

## Features to Add
- [x] ASM Pixel intensity
- [x] Unigram
- [x] Bigram
- [x] Segment Size
- [x] Byte frequency
- [x] Instruction count
- [ ] ~~Byte 4-gram~~
- [x] Random Forest Feature Selection

# Post model creation
- [x] Feature importance
- [x] Voting
- [x] Testing on test data to get accuracy
- [ ] Submit to Kaggle and see result??

## Graphs
- [x] Graphs for different size of data 
- [x] Graphs for different models
- [x] Draw a confusion matrix

# Presentation
- [x] Make presentation
- [x] Practice for presentation to check we can complete in 12 minutes
- [ ] Prepare a demo? -- Can create a web front-end quickly in django which takes the two files and then classifies it. It can also show information about the malware we're dealing with.
- [x] Check beforehand that the laptop can connect to the projector
- [x] Keep the code and terminal open and ready to demo during the presentation

# Make Repo Public
- [x] Make repo public for grading

# Blog (Submission Deadline: 5th night)
- [x] Start writing something!!
- [x] Background about the problem/project
- [x] Background about malware classification
- [x] Issues faced in malware classification in general
- [x] Why this project
- [ ] How we planned it
- [ ] What all we tried
- [ ] Problems we faced working with large datasets
- [ ] What we learnt
- [ ] Reference/Citations
- [ ] How awesome the course was
